Correcting a bug with large numbers and improving some macros

GuillermoCasas opened the pull request:
There was a bug by which very large number of particles were not processed correctly due to the limited range on ints. I also made the macros safer and  optimized some of them a bit.

GuillermoCasas committed:
Correcting a bug with large numbers and improving some macros

There was a bug by which very large number of particles were not processed correctly due to the limited range on ints. I also made the macros safer and  optimized some of them a bit.

maceligueta commented on applications/DEM_application/custom_utilities/inlet.cpp:
```diff
@@ -345,6 +345,7 @@ namespace Kratos {
             int total_mesh_size_accross_mpi_processes = mesh_size_elements; //temporary value until reduction is done
             r_modelpart.GetCommunicator().SumAll(total_mesh_size_accross_mpi_processes);
             const double this_mpi_process_portion_of_inlet_mesh = (double) mesh_size_elements / (double) total_mesh_size_accross_mpi_processes;
+            assert(mp[INLET_NUMBER_OF_PARTICLES] >= 0);
```
Are you sure you want to use assert? It is not used anywhere in Kratos

GuillermoCasas replied:
Ok, I replaced it for something better.

GuillermoCasas committed:
Replacing assert for a KRATOS_ERROR

GuillermoCasas committed:
Merge branch 'master' into debug-inlet-.with-large-numbers

maceligueta commented on applications/DEM_application/custom_utilities/inlet.cpp:
```diff
@@ -662,5 +661,18 @@ namespace Kratos {
+        }
+
+        else {
+            KRATOS_THROW_ERROR(std::runtime_error, "The value of the Model Part variable INLET_NUMBER_OF_PARTICLES is not a positive int: ", num_part_surface_time);
```
I have seen in other PRs that the recommended sintax here is KRATOS_ERROR<<"String" ...
KRATOS_THROW_ERROR is the old style, we have to change some of them.

GuillermoCasas committed:
Moving to new KRATOS_ERROR SYNTAX

GuillermoCasas committed:
Merge remote-tracking branch 'origin/debug-inlet-.with-large-numbers' into debug-inlet-.with-large-numbers

GuillermoCasas committed:
Applying filtering to fluid_fraction

Now it is possible to apply (exponential) time filtering to the fluid fraction. It requires the addition fo the nodal variable FLUID_FRACTION_FILTERED

maceligueta reviewed (changes requested):
Some undesired files were uploaded in the last commit!
TimesPartialRelease, console_output, ...

GuillermoCasas committed:
eliminating temporal files

GuillermoCasas committed:
Eliminating some more temp files

maceligueta commented:
Should we delete this branch?

GuillermoCasas commented:
No, since I am still working on it.
